fire/cli/info.py

```diff
@@ -159,21 +159,32 @@ def info():
 #   Nyeste observation:  1997-07-15 12:00:00
 
 
-def kanonisk_ident(uuid) -> str:
-    try:
-        # TODO: Bør cache både pil, pid og resultater pr UUID, så vi kan reducere opslag
-        pil = firedb.hent_punktinformationtype("IDENT:landsnr")
-        pig = firedb.hent_punktinformationtype("IDENT:GNSS")
-        pid = firedb.hent_punktinformationtype("IDENT:diverse")  # G.I. og G.M.
+# Cache de relevante punktinformationstyper for kanonisk_ident,
+# så de ikke skal slås op ved hvert kald af funktionen
+pit_landsnr = firedb.hent_punktinformationtype("IDENT:landsnr")
+pit_gnssnavn = firedb.hent_punktinformationtype("IDENT:GNSS")
+pit_gi_gs_gm = firedb.hent_punktinformationtype("IDENT:diverse")
+
```

Comment:
Det her er en god ide, der bliver lavet mange overflødige opslag. En mere generisk løsning vil være at bygge det direkte ind i `FireDb.hent_punktinformationtype()`. Jeg forestiller mig noget med en caching dict på `FireDb`, hvor man kan dumpe punktinformationer i takt med at de bliver slået op. `IDENT:xxx` som nøgle. Jeg kan godt lave det i et separat PR hvis det skal være - så støjer det ikke her.

Reply:
Det ville sikkert være fint - jeg kan ikke overskue hvad det egentlig er der sker nede i maskinrummet på SQLAlchemy og hvordan der skal rettes til.

`kanonisk_ident` bør under alle omstændigheder være en API-funktion, men den er jo nem at flytte.

Min kommentar om caching går mere på at jeg har flyttet de tre opslag udenfor funktionen, som globale variable, for at undgå at slå dem op ved hvert funktionskald, så i den forstand bliver de cachet nu (i modsætning til tidligere versioner, hvor de blev kaldt inde i funktionen, hvor man ville have dem liggende som `static`-objekter, der kun blev fyldt op en gang, hvis vi var i C/C++-verdenen.

Reply:
> Min kommentar om caching går mere på at jeg har flyttet de tre opslag udenfor funktionen, som globale variable, for at undgå at slå dem op ved hvert funktionskald, så i den forstand bliver de cachet nu 

Ja, det forstod jeg godt. Mit forslag er ikke kompliceret, det er bare generisk. Jeg prøver lige at se om ikke jeg kan gøre det på de 23 minutter jeg har inden ETRS89 Workshop nr. 1 starter :-)

Reply:
https://github.com/Kortforsyningen/FIRE/pull/156